datafusion/src/physical_plan/array_expressions.rs

```diff
@@ -25,21 +25,91 @@ use std::sync::Arc;
 
 use super::ColumnarValue;
 
-fn array_array(arrays: &[&dyn Array]) -> Result<FixedSizeListArray> {
+fn array_array(arrays: &[&dyn Array]) -> Result<ArrayRef> {
     assert!(!arrays.is_empty());
     let first = arrays[0];
     assert!(arrays.iter().all(|x| x.len() == first.len()));
     assert!(arrays.iter().all(|x| x.data_type() == first.data_type()));
 
     let size = arrays.len();
 
-    let values = concat::concatenate(arrays)?;
```

Comment:
`array_array` is to zip elements at the same index for each array into a FixedSizeArray, therefore not identical to `concatenate` that chain all arrays end to end.